### PR TITLE
JP-2945 Save Pixel Area in extension for subarray region

### LINKED ISSUE #7226: Pixel area extension not cut out for subarray
_Issue [JP-2945](https://jira.stsci.edu/browse/JP-2945) was created on JIRA by [Alicia Canipe](https://jira.stsci.edu/secure/ViewProfile.jspa?name=acanipe):_

The pixel area extension that gets added during the photom step is not extracting the subarray (the full 2048x2048 pixel area map is being appended). Example for SUB320 attached.

### PR DESCRIPTION
**Pull request by jemorrison**
Resolves [JP-2945](https://jira.stsci.edu/browse/JP-2945)

Closes #7226 

This PR addresses saves the pixel area for the subarray region. The region saved now is set by input datas meta.subarray.xystart: meta.subarray.xysize+1

**Checklist for maintainers**
- [x] added entry in `CHANGES.rst` within the relevant release section
- [x] updated or added relevant tests
- [ ] updated relevant documentation
- [x] added relevant milestone
- [x] added relevant label(s)
- [X ] ran regression tests, post a link to the Jenkins job below.
Two expected differences. Area map for NIRCAM subarray size of subarray
https://plwishmaster.stsci.edu:8081/job/RT/job/JWST-Developers-Pull-Requests/455/
      [How to run regression tests on a PR](https://github.com/spacetelescope/jwst/wiki/Running-Regression-Tests-Against-PR-Branches)
- [ ] Make sure the JIRA ticket is [resolved properly](https://github.com/spacetelescope/jwst/wiki/How-to-resolve-JIRA-issues)

### CODE REVIEW

**Comment by jemorrison** [comment]
@hbushouse 
First attempt on this PR. This works for NIRCAM data. But I am not sure any changes NIRSPEC or data that is MULTISLITMODEL are needed.  I wanted your opinion on this. I was thinking NO.

File: jwst/photom/photom.py
Comment (hbushouse) [inline jwst/photom/photom.py:1035]: The xstart/ystart values are 1-indexed, so you'll need to convert them to 0-indexed before using them as slices. So it should be `[xy]start - 1`. And it needs to end at `ystart+ysize-1`.

**Comment by hbushouse** [comment]
Looks like the `create_input` function in /jwst/photom/tests/test_photom.py needs to be updated to populate the x/ystart and x/ysize attributes in the data models it creates. Their absence is what's killing the CI tests, due to at least one unit test failing.

**Comment by jemorrison** [comment]
Looking into some unexpected results from the regression tests

**Review by hbushouse — APPROVED** [review]
Code updates look good. Just waiting for CI results before merging.

**Pull request merged**
